moveit_simple/test/moveit_simple.cpp

```diff
@@ -468,4 +468,23 @@ TEST(MoveitSimpleTest, collision)
   EXPECT_TRUE(robot2.isInCollision(joint3));
   EXPECT_THROW(robot2.execute(TRAJECTORY_NAME, true), moveit_simple::CollisionDetected);
 }
+
+TEST(MoveitSimpleTest, Singularity)
+{
+  moveit_simple::Robot robot(ros::NodeHandle(), "robot_description", "manipulator");
+  ros::Duration(2.0).sleep();  //wait for tf tree to populate
+
+  std::vector<trajectory_msgs::JointTrajectoryPoint> points;
+
+  std::vector<double>joint1(6,0);
+  std::vector<double>joint2(6,0);
+  joint1[4] = M_PI/2;
```

Comment:
Whenever 4 & 6 align, 6 parallel to the world above axis 1, whenver the "elbow" is straight, any configs at the "extent" of the reach.